tests/deepsparse/loggers/metric_functions/template_functions/test_template_functions.py

```diff
@@ -12,11 +12,163 @@
+        ),
+        (
+            "sentiment_analysis",
+            "sentiment_analysis",
+            {"sequences": ["the food tastes great", "the food tastes bad"]},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/sentiment_analysis_logs_2.txt",  # noqa E501
+        ),
+        (
+            "sentiment_analysis",
+            "sentiment_analysis",
+            {
+                "sequences": [
+                    ["the food tastes great"],
+                    ["the food tastes bad"],
+                ]
+            },
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/sentiment_analysis_logs_3.txt",  # noqa E501
+        ),
+        (
+            "zero_shot_text_classification",
+            "zero_shot_text_classification",
+            {"sequences": "the food tastes great", "labels": ["politics", "food"]},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/zero_shot_text_classification_logs_1.txt",  # noqa E501
+        ),
+        (
+            "zero_shot_text_classification",
+            "zero_shot_text_classification",
+            {
+                "sequences": ["the food tastes great", "the government is corrupt"],
+                "labels": ["politics", "food"],
+            },
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/zero_shot_text_classification_logs_2.txt",  # noqa E501
+        ),
+        (
+            "token_classification",
+            "token_classification",
+            {"inputs": "the food tastes great"},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/token_classification_logs_1.txt",  # noqa E501
+        ),
+        (
+            "token_classification",
+            "token_classification",
+            {"inputs": ["the food tastes great", "the food tastes bad"]},
+            "tests/deepsparse/loggers/metric_functions/template_functions/template_logs/token_classification_logs_2.txt",  # noqa E501
+        ),
+    ],
+)
+@mock_engine(rng_seed=0)
+def test_group_name(mock_engine, group_name, pipeline_name, inputs, expected_logs):
+    yaml_config = """
+    loggers:
+        list_logger:
+            path: tests/deepsparse/loggers/helpers.py:ListLogger
+    add_predefined:
+    - func: {group_name}"""
+
+    if pipeline_name == "zero_shot_text_classification":
+        pipeline = Pipeline.create(
+            pipeline_name,
+            logger=logger_from_config(
+                config=yaml_config.format(group_name=group_name),
+                pipeline_identifier=pipeline_name,
+            ),
+        )
+    else:
+        pipeline = Pipeline.create(
+            pipeline_name, logger=yaml_config.format(group_name=group_name)
+        )
+
+    pipeline(**inputs)
+    # let's sleep one seconds so all the logs get collected
+    time.sleep(1)
```

Comment:
do we need to sleep for this long? additionally are the loggers running wrapped in async? adding a mandatory second for each test will add up